src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReference.Snapshot.cs

```diff
@@ -34,16 +34,20 @@ internal sealed class Snapshot : PortableExecutableReference, ISupportTemporaryS
             private readonly VisualStudioMetadataReferenceManager _provider;
             private readonly Lazy<DateTime> _timestamp;
             private Exception _error;
+            private FileChangeTracker _fileChangeTrackerOpt;
 
-            internal Snapshot(VisualStudioMetadataReferenceManager provider, MetadataReferenceProperties properties, string fullPath)
+            internal Snapshot(VisualStudioMetadataReferenceManager provider, MetadataReferenceProperties properties, string fullPath, FileChangeTracker fileChangeTrackerOpt)
                 : base(properties, fullPath)
             {
                 Contract.Requires(Properties.Kind == MetadataImageKind.Assembly);
                 _provider = provider;
+                _fileChangeTrackerOpt = fileChangeTrackerOpt;
 
                 _timestamp = new Lazy<DateTime>(() => {
                     try
                     {
+                        _fileChangeTrackerOpt?.EnsureSubscription();
```

Comment:
>_fileChangeTrackerOpt?.EnsureSubscription(); [](start = 24, length = 44)

Do we want to track the file if we can't get the file stamp?

Reply:
Probably? If the problem was transient, but an update would fix it, then yes?

Reply:
Makes sense.